interface/modules/custom_modules/oe-module-faxsms/src/Controller/EtherFaxActions.php

```diff
@@ -0,0 +1,525 @@
+        $this->appKey = $credentials['appKey'];
+        $this->appSecret = $credentials['appSecret'];
+        $this->serverUrl = (isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] === 'on' ?
+                "https" : "http") . "://" . $_SERVER['HTTP_HOST'];
+        $this->uriDir = $this->serverUrl . $this->uriDir;
+
+        return $credentials;
+    }
+
+    public function fetchReminderCount(): bool|string
+    {
+        $cnt = $this->client->getUnreadFaxCount();
+
+        return json_encode($cnt);
+    }
+
+        /**
+     * @return string
+     */
+    public function faxProcessUploads(): string
+    {
+        if (!empty($_FILES)) {
+            $name = $_FILES['fax']['name'];
+            $ext = $_FILES['fax']['ext'];
+            $tmp_name = $_FILES['fax']['tmp_name'];
+        } else {
+            return 'Error';
+        }
+        if (!file_exists($this->baseDir . '/send')) {
+            mkdir($this->baseDir . '/send', 0777, true);
+        }
+        // add to fax queue
+        ['basename' => $basename, 'dirname' => $dirname] = pathinfo($tmp_name);
+        $filepath = $this->baseDir . "/send/" . $name;
+
+        move_uploaded_file($tmp_name, $filepath);
+
+        return $filepath;
+    }
+
+    /**
+     * @return string
+     */
+    public function sendSMS(): string
+    {
+        // dummy function
+        return text("Not implemented");
+    }
+
+    /**
+     * @return mixed|string
+     */
+    public function sendFax(): string|bool
+    {
+        if (!$this->authenticate()) {
+            return xlt('Error: Authentication Service Denies Access.');
+        }
+        $isContent = $this->getRequest('isContent');
+        $file = $this->getRequest('file');
+        $docid = $this->getRequest('docid');
+        $mime = $this->getRequest('mime');
+        $phone = $this->getRequest('phone');
+        $isDocuments = (int)$this->getRequest('isDocuments');
+        $isQueue = $this->getRequest('isQueue');
+        $comments = $this->getRequest('comments');
+        $content = '';
+        $phone = $this->formatPhone($phone);
+        $from = $this->formatPhone($this->credentials['phone']);
+        $status = [];
+        $user = $this::getLoggedInUser();
+        $csid = $user['facility'];
+        if (!$isContent) {
+            $file = str_replace("file://", '', $file);
+            $file = str_replace("\\", "/", realpath($file)); // normalize requested path
+            if (!$file) {
+                return xlt('Error: No content');
+            }
+        }
+        ['basename' => $basename, 'dirname' => $dirname] = pathinfo($file);
+        if ($isContent) {
+            $content = $file;
+            $file = 'report-' . $GLOBALS['pid'] . '.pdf';
+        }
+        if ($isDocuments) {
+            $obj = new \C_Document();
+            $file = $obj->retrieve_action("", $docid, true, true, true);
+            // is it encrypted
+            if ($this->crypto->cryptCheckStandard($content)) {
```

Comment:
cleaner way (in case there is a one in a thousand chance or so that a document starts with 001-006) would be:
```
if ((new Document($docid))->get_encrypted() == 1)  {
```

Reply:
retrieve will decrypt if encrypted. I don't need to know if doc was encrypted just dbl check ensure content not encrypted.

Reply:
(new Document($docId))->get_data() will also accomplish this behavior as it decrypts the document automatically unless you pass in the false flag.  I'm interested to know this 001-006 stuff.  Is that in the cryptCheckStandard?  Just want to make sure I don't hit that in the future.

Reply:
Regarding 001-006, this is prefixed in the encrypted data (can see this by looking at encrypted file in text editor or less command). Check out top of CryptoGen class text on descriptions of the $encryptionVersion variable ($keyVersion also helpful).  These are to support backward compatibility of prior encryption schemes. Encrypted items will always have 006 since this is most recent encryption scheme. Items to be decrypted can be any of these depending on when they were encrypted. These things allow us to fix/improve/modernize encryption without breaking decryption using the older methods.

Reply:
Also, to clarify, since all encrypted data has to be prefixed with 001-006, this is basically what cryptCheckStandard will check for (line 129 of CryptoGen). Ideally, this should only be done if there is no other way to figure out if the file was encrypted (for example, patient documents are flagged in database when encrypted so would use that instead; but the logs that are encrypted in documents subdirectories, there is no flag stored anywhere, so need the cryptCheckStandard  function).